### PR TITLE
[CIS-625] Separate text and attachment cells

### PR DESCRIPTION
**Pull request by b-onc**
WIP

### CODE REVIEW

File: Sources/StreamChatUI/UIConfig+MessageList.swift [hunk 1]
```diff
@@ -7,6 +7,12 @@ import UIKit.UIImage
 
 public extension _UIConfig {
     struct MessageListUI {
+        public var cellReuseIdentifierForMessage: (_ChatMessageGroupPart<ExtraData>, _UIConfig<ExtraData>) -> String? = { _, _ in
+            nil
+        }
+        
+        public var registerCustomCellClass: (ChatMessageListCollectionView) -> Void = { _ in }
+
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/UIConfig+MessageList.swift:15]: I think those should live variables in `ChatMessageListVC` directly.

**Review by VojtaStavik — COMMENTED** [review]
Looks great IMO 👏

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift [hunk 2]
```diff
@@ -168,6 +175,22 @@ open class _ChatMessageListVC<ExtraData: ExtraDataTypes>: ViewController,
             self.scrollToMostRecentMessageIfNeeded()
         }
     }
+    
+    public func cellReuseIdentifierForMessage(_ message: _ChatMessageGroupPart<ExtraData>) -> String {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift:179]: `open`

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageBubbleView.swift [hunk 3]
```diff
@@ -52,7 +44,7 @@ open class _ChatMessageBubbleView<ExtraData: ExtraDataTypes>: View, UIConfigProv
     
     public private(set) lazy var borderLayer = CAShapeLayer()
 
-    private var layoutConstraints: [LayoutOptions: [NSLayoutConstraint]] = [:]
+    fileprivate var layoutConstraints: [LayoutOptions: [NSLayoutConstraint]] = [:]
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageBubbleView.swift:47]: let's make this `public` too, we will probably remove it soon, but let's keep it as flexible as we can for now.

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift [hunk 4]
```diff
@@ -282,3 +282,22 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: View, UIConfigPro
...
+            return messageBubbleView
+        } else {
+            _messageAttachmentBubbleView = uiConfig
+                .messageList
+                .messageContentSubviews
+                .attachmentBubbleView.init()
+                .withoutAutoresizingMaskConstraints
+            return _messageAttachmentBubbleView!
+        }
+    }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:302]: What's happening here?

**Review by VojtaStavik — APPROVED** [review]
:shipit: 🚀

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift (hunk 4 above)
Comment (b-onc) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:302]: `messageBubbleView` is a lazy property in superclass, so subclass cannot override it with a stored (lazy) property. I've overcame that limitation by basically emulating the `lazy` property with a (non-overridden) stored property

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageCollectionViewCell.swift [hunk 5]
```diff
@@ -91,3 +91,38 @@ class СhatOutgoingMessageCollectionViewCell<ExtraData: ExtraDataTypes>: _СhatM
...
+                .messageList
+                .messageAttachmentContentView
+                .init()
+                .withoutAutoresizingMaskConstraints
+            return _messageAttachmentContentView!
+        }
+    }
+}
+
+class СhatIncomingMessageAttachmentCollectionViewCell<ExtraData: ExtraDataTypes>:
```
[the remark this answers is not in the document]
Comment (b-onc) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageCollectionViewCell.swift:114]: Swiftformatter works but apparently doesn't fix these, interesting....

**Pull request merged**